pkg/apiclient/apiclient.go

```diff
@@ -383,10 +389,21 @@ func (c *client) newConn() (*grpc.ClientConn, io.Closer, error) {
 	var dialOpts []grpc.DialOption
 	dialOpts = append(dialOpts, grpc.WithPerRPCCredentials(endpointCredentials))
 	dialOpts = append(dialOpts, grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(MaxGRPCMessageSize)))
+
+	ctx := context.Background()
+
+	if len(c.AdditionalHeaders) > 0 {
+		for _, kv := range c.AdditionalHeaders {
+			if len(strings.Split(kv, ":"))%2 == 1 {
```

Comment:
minor - I don't think you should bother with this check - unless you plan to write a test for it